File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/views/login/AccountsView.java
package app.editors.manager.mvp.views.login;

import moxy.viewstate.strategy.OneExecutionStateStrategy;
import moxy.viewstate.strategy.StateStrategyType;

import java.util.List;

import app.editors.manager.mvp.models.account.AccountsSqlData;
import app.editors.manager.mvp.views.base.BaseView;

@StateStrategyType(OneExecutionStateStrategy.class)
public interface AccountsView extends BaseView {
    void onAccountLogin();
    void onUsersAccounts(List<AccountsSqlData> accounts);
    void onAccountDelete(int position);
    void onSignIn(String portal, String login);
    void showWaitingDialog();
    void onWebDavLogin(AccountsSqlData mAccountClickedItem);
}

File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/views/login/AccountsView.kt
package app.editors.manager.mvp.views.login

import app.documents.core.account.CloudAccount
import app.editors.manager.mvp.views.base.BaseView
import moxy.viewstate.strategy.OneExecutionStateStrategy
import moxy.viewstate.strategy.StateStrategyType

@StateStrategyType(OneExecutionStateStrategy::class)
interface AccountsView : BaseView {
    fun onAccountLogin()
    fun onUsersAccounts(accounts: MutableList<CloudAccount>)
    fun onAccountDelete(position: Int)
    fun onSignIn(portal: String, login: String)
    fun showWaitingDialog()
    fun onWebDavLogin(account: CloudAccount)
    fun onOneDriveLogin(account: CloudAccount)
    fun onDropboxLogin(account: CloudAccount)
    fun onGoogleDriveLogin(account: CloudAccount)
}